### PR TITLE
doc: Add download page

### LINKED ISSUE #213: Add a download page
It is empty currently: https://rust.iceberg.apache.org/download.html

### PR DESCRIPTION
**Pull request by liurenjie1024**
Close #213.

Add download page for book.

### CODE REVIEW

File: website/src/download.md [hunk 1]
```diff
@@ -17,3 +17,45 @@
   ~ under the License.
 -->
 
+# Apache Iceberg-Rust™ Downloads
+
+The official Apache Iceberg-Rust releases are provided as source artifacts.
+
+## Releases
+
+The latest source release is [0.2.0](https://archive.apache.org/dist/iceberg/iceberg-rust-0.2.0/apache-iceberg-rust-0.2.0-src.tar.gz) ([asc](https://archive.apache.org/dist/iceberg/iceberg-rust-0.2.0/apache-iceberg-rust-0.2.0-src.tar.gz.asc),
```
Comment (Xuanwo) [inline website/src/download.md:26]: We must use link that point to `https://www.apache.org/dyn/closer.lua/iceberg/iceberg-rust-0.2.0/apache-iceberg-rust-0.2.0-src.tar.gz?action=download`.

File: website/src/download.md [hunk 2]
```diff
@@ -17,3 +17,45 @@
...
+[sha512](https://archive.apache.org/dist/iceberg/iceberg-rust-0.2.0/apache-iceberg-rust-0.2.0-src.tar.gz.sha512)).
```
Comment (Xuanwo) [inline website/src/download.md:27]: asc and sha512 must point to `https://downloads.apache.org`

File: website/src/download.md [hunk 3]
```diff
@@ -17,3 +17,45 @@
...
+
+For older releases, please check the [archive](https://archive.apache.org/dist/iceberg/).
+
+## Notes
+
+* When downloading a release, please verify the OpenPGP compatible signature (or failing that, check the SHA-512); these should be fetched from the main Apache site.
+* The KEYS file contains the public keys used for signing release. It is recommended that (when possible) a web of trust is used to confirm the identity of these keys.
+* Please download the [KEYS](https://archive.apache.org/dist/iceberg/KEYS) as well as the .asc signature files.
```
Comment (Xuanwo) [inline website/src/download.md:35]: KEYS must download from `https://downloads.apache.org/iceberg/KEYS`

File: website/src/download.md [hunk 4]
```diff
@@ -17,3 +17,45 @@
   ~ under the License.
 -->
 
+# Apache Iceberg-Rust™ Downloads
```
Comment (Xuanwo) [inline website/src/download.md:20]: Please don't add `TM` upon rust.

We should use `Apache Iceberg™ Rust`.

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Review by Fokko — APPROVED** [review]
Thanks for adding this @Xuanwo 🙌

**Pull request merged**